src/ProblemDetailsNotFoundHandler.php

```diff
@@ -0,0 +1,60 @@
+<?php
+namespace Zend\ProblemDetails;
+
+use Interop\Http\ServerMiddleware\DelegateInterface;
+use Interop\Http\ServerMiddleware\MiddlewareInterface as ServerMiddlewareInterface;
+use Negotiation\Negotiator;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Zend\Stratigility\Delegate\CallableDelegateDecorator;
+
+class ProblemDetailsNotFoundHandler implements ServerMiddlewareInterface
+{
+    /**
+     * @var ProblemDetailsResponseFactory
+     */
+    private $responseFactory;
+
+    /**
+     * @param ProblemDetailsResponseFactory $responseFactory Factory to create a response to
+     *     update and return when returning an 404 response.
+     */
+    public function __construct(ProblemDetailsResponseFactory $responseFactory = null)
+    {
+        $this->responseFactory = $responseFactory ?: new ProblemDetailsResponseFactory();
```

Comment:
As we required PHP7.1 we can have here: 
```php
 $responseFactory ?? new ProblemDetailsResponseFactory();
```

Not sure if it is more efficient or it doesn't matter at all?

Reply:
Makes no different. `?:` is consistent with `ProblemDetailsMiddleware::__construct()`

Reply:
`??` is mostly useful if the variable referenced might not be defined; it adds an implicit `isset`. Since we know it's defined (it's in the constructor), `?:` makes sense here.